client/src/controllers/ProgressController.test.js

```diff
@@ -0,0 +1,106 @@
+import { Application } from '@hotwired/stimulus';
+import { ProgressController } from './ProgressController';
+
+jest.useFakeTimers();
+
+const flushPromises = () => new Promise(setImmediate);
+describe('ProgressController', () => {
+  // form submit is not implemented in jsdom
+  const mockSubmit = jest.fn((e) => e.preventDefault());
+
+  beforeEach(() => {
+    document.body.innerHTML = `
+      <form id="form">
+        <button
+          id="button"
+          type="submit"
+          class="button button-longrunning"
+          data-controller="w-progress"
+          data-action="w-progress#activate"
```

Comment:
```suggestion
          data-action="w-progress#activate"
          data-w-progress-active-value="Loading"
```

So that existing tests pass.